### PR TITLE
Complete blog example

### PR DESCRIPTION
**Pull request by keshon**
I made blog example complete. Few comments about changes:
- Added missing routes.
- Rewritten templates (old are gone).
- Small changes in sql scheme.
- Different visual style (using Marx classless framework with some custom touches) .

Few things to consider:
- I have almost zero experience in Nim.
- I'm not entirely familiar with code guidelines (and I use 4 spaces instead of 2).

### CODE REVIEW

**Comment by xflywind** [comment]
Thanks for your effores.
Please use `nimpretty --indent:2` to change four space to two space.
```
C:\WINDOWS\system32>nimpretty --help
nimpretty - Nim Pretty Printer Version 0.2
  (c) 2017 Andreas Rumpf
Usage:
  nimpretty [options] file.nim
Options:
  --out:file            set the output file (default: overwrite the input file)
  --indent:N[=0]        set the number of spaces that is used for indentation
                        --indent:0 means autodetection (default behaviour)
```

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+*.exe
```
Comment (xflywind) [inline .gitignore:1]: Please add `.gitignore` to .ignore files.

Comment (keshon) [reply .gitignore:1]: I added gitignore to itself but I don't know how to remove it from my github repo.
I used nimpretty on all my files.

**Comment by xflywind** [comment]
Thanks! The code looks good to me.

File: .gitignore (hunk 1 above)
Comment (xflywind) [reply .gitignore:1]: Ok. I will delete them.

**Comment by xflywind** [comment]
Maybe you can explain how to run the projects in readme.(first initialzie the database or so) `not an error` is confusing for beginner. :)

**Comment by xflywind** [comment]
image

I ran into some errors. Do you have some ideas?

**Comment by xflywind** [comment]
I have tried it. The project looks really nice. I will look into some codes now.

**Comment by keshon** [comment]
Please could you try to run the example in Incognito/private mode to be sure old cookies don't conflict with the new one?С уважением, Иннокентий. ---- On Вс, 13 сен 2020 16:59:01 +0300  notifications@github.com  wrote ----

I ran into some errors. Do you have some ideas?

**Comment by keshon** [comment]
I can extend readme file to give better understanding. Give me few days.The db init is automatic: if db file doesn't exist it will be created automatically on first request. I will reflect it in readme.С уважением, Иннокентий. ---- On Вс, 13 сен 2020 16:54:34 +0300  notifications@github.com  wrote ----
Maybe you can explain how to run the projects in readme.(first initialzie the database or so) not an error is confusing for beginner. :)

**Comment by keshon** [comment]
I will take a look at it as well. I remember this error but don't remember how I managed to fix it. Just got away from PC already.С уважением, Иннокентий. ---- On Вс, 13 сен 2020 17:06:37 +0300  notifications@github.com  wrote ----
I have tried it. The project looks really nice. I will look into some codes now.

**Comment by xflywind** [comment]
All right. I think I have solved that problem.

**Comment by keshon** [comment]
> All right. I think I have solved that problem.

If I'm not mistaken this error might occur because you've probably used old blog.db database file. But since you've resolved it already it doesn't matter :)

File: examples/blog/initdb.nim [hunk 2]
```diff
@@ -3,14 +3,14 @@ import db_sqlite, os, strutils, logging
...
-    db.exec(sql(line.strip))
-  db.close()
-  logging.info("Initialized the database.")
+proc initDb*() =
+  if not fileExists(consts.dbPath):
+    let
+      db = open(consts.dbPath, "", "", "")
+      schema = readFile(schemaPath)
+    for line in schema.split(";"):
+      if line == "\c\n":
```
Comment (xflywind) [inline examples/blog/initdb.nim:12]: I think this line should be changed to this, or it cause errors when first initial database. You can delete db file to reproduce it.
```nim
if line == "\c\n" or line == "\n":
   continue
```

Comment (xflywind) [reply examples/blog/initdb.nim:12]: d:\QQPCMgr\Desktop\tp\examples\blog>nim c -r app.nim
Hint: used config file 'C:\Users\blue\.choosenim\toolchains\nim-1.2.6\config\nim.cfg' [Conf]
Hint: used config file 'd:\QQPCMgr\Desktop\tp\config.nims' [Conf]
Hint: used config file 'd:\QQPCMgr\Desktop\tp\examples\blog\config.nims' [Conf]
Hint: 15808 LOC; 0.040 sec; 17.852MiB peakmem; Debug build; proj: d:\QQPCMgr\Desktop\tp\examples\blog\app.nim; out: d:\QQPCMgr\Desktop\tp\examples\blog\app.exe [SuccessX]
Hint: d:\QQPCMgr\Desktop\tp\examples\blog\app.exe  [Exec]
d:\QQPCMgr\Desktop\tp\examples\blog\app.nim(8) app
d:\QQPCMgr\Desktop\tp\examples\blog\initdb.nim(16) initDb
C:\Users\blue\.choosenim\toolchains\nim-1.2.6\lib\impure\db_sqlite.nim(207) exec
C:\Users\blue\.choosenim\toolchains\nim-1.2.6\lib\impure\db_sqlite.nim(142) dbError
Error: unhandled exception: not an error [DbError]
Error: execution of an external program failed: 'd:\QQPCMgr\Desktop\tp\examples\blog\app.exe '

Comment (keshon) [reply examples/blog/initdb.nim]: I added your suggestion but still wasn't able to reproduce strange behavior you have. Did you test on 1.2.6 nim?
I also fixed DROP statement in schema file (i made a typo).

**Comment by keshon** [comment]
Ok I will take a look tomorrow.С уважением, Иннокентий. ---- On Вс, 13 сен 2020 17:51:41 +0300  notifications@github.com  wrote ----
@xflywind commented on this pull request.



In examples/blog/initdb.nim:
> -  let
-    db = open(consts.dbPath, "", "", "")
-    schema = readFile(schemaPath)
-  for line in schema.split(";"):
-    if line == "\c\n":
-      continue
-    db.exec(sql(line.strip))
-  db.close()
-  logging.info("Initialized the database.")
+proc initDb*() =
+  if not fileExists(consts.dbPath):
+    let
+      db = open(consts.dbPath, "", "", "")
+      schema = readFile(schemaPath)
+    for line in schema.split(";"):
+      if line == "\c\n":

d:\QQPCMgr\Desktop\tp\examples\blog>nim c -r app.nim
Hint: used config file 'C:\Users\blue.choosenim\toolchains\nim-1.2.6\config\nim.cfg' [Conf]
Hint: used config file 'd:\QQPCMgr\Desktop\tp\config.nims' [Conf]
Hint: used config file 'd:\QQPCMgr\Desktop\tp\examples\blog\config.nims' [Conf]
Hint: 15808 LOC; 0.040 sec; 17.852MiB peakmem; Debug build; proj: d:\QQPCMgr\Desktop\tp\examples\blog\app.nim; out: d:\QQPCMgr\Desktop\tp\examples\blog\app.exe [SuccessX]
Hint: d:\QQPCMgr\Desktop\tp\examples\blog\app.exe  [Exec]
d:\QQPCMgr\Desktop\tp\examples\blog\app.nim(8) app
d:\QQPCMgr\Desktop\tp\examples\blog\initdb.nim(16) initDb
C:\Users\blue.choosenim\toolchains\nim-1.2.6\lib\impure\db_sqlite.nim(207) exec
C:\Users\blue.choosenim\toolchains\nim-1.2.6\lib\impure\db_sqlite.nim(142) dbError
Error: unhandled exception: not an error [DbError]
Error: execution of an external program failed: 'd:\QQPCMgr\Desktop\tp\examples\blog\app.exe '

File: examples/blog/views.nim [hunk 3]
```diff
@@ -54,22 +63,83 @@ proc register*(ctx: Context) {.async.} =
...
+    resp htmlResponse(registerPage(ctx, "Register"))
 
   db.close()
+
+# /
+proc read*(ctx: Context) {.async.} =
+  let db = open(consts.dbPath, "", "", "")
+  var posts: seq[seq[string]] = @[]
+  for x in db.fastRows(sql"SELECT * FROM posts"):
+    echo x
```
Comment (xflywind) [inline examples/blog/views.nim:94]: comment echo or use logging

**Comment by xflywind** [comment]
I have reviewed all the codes. They seem to work fine. I think it needs a simple description or a sceenshoot in  readme before merging. Then you could post the readme in [reddit](https://www.reddit.com/r/nim) or not. You can also add your name in readme such as (Designed by @keshon). Thanks!

**Comment by keshon** [comment]
> I have reviewed all the codes. They seem to work fine. I think it needs a simple description or a sceenshoot in readme before merging. Then you could post the readme in [reddit](https://www.reddit.com/r/nim) or not. You can also add your name in readme such as (Designed by @keshon). Thanks!

I updated Readme.md take a look at it please. English is not my native to be honest so you are free to edit the file.
I did make a simple screenshot but dont know how to attach it to Readme.md properly (I mean I don't know where do I store the image). Attaching the screenshot here.
preview

As for reddit - I'm kinda speechless guy when it comes to audience, will think about it.

**Comment by xflywind** [comment]
Amazing work! Thanks!

**Pull request merged**